elastic/elastic-agent
# Fix (*artifact.Config).Unpack not preserving not overridden values

## What does this PR do?

(*artifact.Config).Unpack was creating a new instance of Config and manually coping all values from the original to the new one. New values added to config, RetrySleepInitDuration in this case, were neither maintaining its default value nor being updated with the new value. Now it uses reflection to ensure all values are correctly copied.

## Why is it important?

When applying a new `agent.download` config some of the default values were set to the zero value and even if they'd come on the new fleet config, they'd be ignored.

## Checklist




## How to test this PR locally
Try any of the following:
 - Copy the test `TestConfig_Unpack` to main and see it failing.
 - Set the `retry_sleep_init_duration` in the agent's policy, enrol it to fleet and check it's not set.
 - Reproduce the issue described in #3914

## Related issues

- Closes #3914

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
